### ISSUE TITLE
How to prepare train data and how to train DeblurGAN?

### ISSUE THREAD

**Issue by hrbspider** [KupynOrest/DeblurGAN#18]
I tried to train DeblurGAN but failed. I dont know how to prepare train data and val data and the meaning of some options,such as 'AtoB'.
I generated blured images using the code blur_image.py.
The train command:
python train.py --dataroot ../../deblurPwLine/ --continue_train

dataroot is organized as this:
-deblurPwLine
   --train "stores some sharp images"
   --val "stores some sharp images"

**Comment by KupynOrest** [comment]
There are few options and few choices of dataset (Aligned, Single etc.) To create aligned dataset you would need to run combine_A_and_B script to create image pairs, otherwise, you need to change the type of the dataset, I would update to instructions soon

**Comment by carlblocking** [comment]
Can you provide more information about it? @KupynOrest

**Comment by carlblocking** [comment]
Can you provide more information about it? @KupynOrest

**Comment by YangShuguang** [comment]
when I generated blured images using the code blur_image.py. I found a lot of picture become Black block. It seems that all the pixel value become zero. how I solve the problem? @hrbspider  @KupynOrest

**Comment by pengjingg** [comment]
@YangShuguang Use scipy.misc.imsave() to save the generated blurry image instead of cv2.imwrite()

**Comment by pengjingg** [comment]
@YangShuguang You should use either OpenCV or Scipy while reading and writing images; use one to read images and another to write images would cause black block output.

**Comment by YangShuguang** [comment]
@pengjingg.  are you sure? I'll try your way.  now my solution is to modify the "self.max_len / (self.iters - 1))" to "self.max_len / float(self.iters - 1))" (generate_trajectory.py 80 lines). and modify "self.fraction = [1 / 100, 1 / 10, 1 / 2, 1]'' to ''self.fraction = [1/float(100), 1/float(10), 1/float(2), 1]'' (generate_PSF.py 19 lines ). it seems to work

**Comment by pengjingg** [comment]
@YangShuguang You are right. My way just solved problems for some images but won't work out for others(strange). Thanks a lot
